scripts/services/map_manager.gd

```diff
@@ -43,20 +49,26 @@ func get_maps_page(page_number, page_size):
     var map_key_list = self.maps.keys()
     map_key_list.sort()
 
-    if index_end > map_key_list.size():
-        index_end = map_key_list.size()
+    if index_end > _get_maps_count():
```

Comment:
Please stick to convention to explicitly use 'self.*', so that it is clear, that we are using a method from this class, and not a function from global scope.